presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushProjectionIntoTableScan.java

```diff
@@ -116,7 +120,9 @@ public Result apply(ProjectNode project, Captures captures, Context context)
         Optional<ProjectionApplicationResult<TableHandle>> result = metadata.applyProjection(context.getSession(), tableScan.getTable(), connectorPartialProjections, assignments);
 
         if (result.isEmpty()) {
-            return Result.empty();
+            return pushAssignmentsIntoTableScan(project, tableScan)
+                    .map(Result::ofPlanNode)
```

Comment:
This looks like rule-in-rule.

OTOH, it feels like then problem should be handled by `PushProjectionIntoTableScan` somehow, so i kept it here at least for now.